labgrid/protocol/backgroundprocessprotocol.py

```diff
@@ -0,0 +1,67 @@
+import abc
+
+
+class BackgroundProcessProtocol(abc.ABC):
+    """
+    Abstract class providing the BackgroundProcessProtocol interface. It allows running commands
+    as background processes. Running multiple commands concurrently may work depending on the
+    driver.
+
+    Example usage:
+
+    >>> from labgrid.util.timeout import Timeout
+    >>> with Timeout(300.0) as timeout:
+    >>>     process = driver.run_as_background_process('dd if=/dev/random bs=1M count=100')
+    >>>     while driver.poll_background_process(process) is None and not timeout.expired:
+    >>>         time.sleep(min(timeout.remaining, 1.0))
+    >>>         stdout, stderr = driver.read_from_background_process(process, timeout=timeout.remaining)
```

Comment:
Here we don't want to block for the remaining time
```suggestion
    >>>         stdout, stderr = driver.read_from_background_process(process)
```